presto-iceberg/src/test/java/com/facebook/presto/iceberg/hive/TestIcebergHiveStatistics.java

```diff
@@ -332,6 +340,50 @@ public void testHiveStatisticsMergeFlags()
         assertEquals(columnStatistics.get("v").getDataSize(), Estimate.of(11));
     }
 
+    @DataProvider(name = "pushdownFilterEnabled")
+    public Object[][] pushdownFilterPropertyProvider()
+    {
+        return new Object[][] {
+                {true, true},
+                {true, false},
+                {false, true},
+                {false, false},
+        };
+    }
+
+    @Test(dataProvider = "pushdownFilterEnabled")
+    public void testPredicateOnlyColumnInStatisticsOutput(boolean pushdownFilterEnabled, boolean partitioned)
+    {
+        assertQuerySucceeds(format("CREATE TABLE scanFilterStats (i int, j int, k int)%s", partitioned ? " WITH (partitioning = ARRAY['j'])" : ""));
+        assertUpdate("INSERT INTO scanFilterStats VALUES (1, 2, 3), (3, 4, 5), (5, 6, 7)", 3);
+        Session session = Session.builder(getSession())
+                .setCatalogSessionProperty("iceberg", PUSHDOWN_FILTER_ENABLED, Boolean.toString(pushdownFilterEnabled))
+                .build();
+        try {
+            TableStatistics est = getScanStatsEstimate(session, "SELECT k from scanFilterStats WHERE j > 2 AND i = 3");
+            assertEquals(est.getColumnStatistics().size(), 3);
+        }
+        finally {
+            getQueryRunner().execute("DROP TABLE scanFilterStats");
+        }
```

Comment:
Seems that the session property `iceberg.pushdown_filter_enabled` not working. I believe it is missed in `AbstractTestingPrestoClient.toClientSession(...)`.

And furthermore, if `iceberg.pushdown_filter_enabled` works, here we would only get an exception, as pushdown_filter_enabled currently do not support java worker. So maybe we should figure out another way to test such scenarios.

Reply:
> I believe it is missed in AbstractTestingPrestoClient.toClientSession(...).

I'm not sure what you mean here. Could you elaborate?

> And furthermore, if iceberg.pushdown_filter_enabled works, here we would only get an exception when it's true

`pushdown_filter_enabled` only throws an error once the splits are generated. In this test, we're not running the query. Just generating the query plan, so we shouldn't see an error thrown here. It should work fine in pure Java

Reply:
> I'm not sure what you mean here. Could you elaborate?

We can make a breakpoint in method `IcebergFilterPushdown.optimize(...)` to observe the validity of session property `iceberg.pushdown_filter_enabled`. Seems it was missed after `session.beginTransactionId(...)` and `AbstractTestingPrestoClient.toClientSession(...)` in method `TestIcebergHiveStatistics.getScanStatsEstimate(...)`.

> In this test, we're not running the query. Just generating the query plan, so we shouldn't see an error thrown. It should work fine in pure Java

It seems that `getQueryRunner().createPlan(txnSession, sql, WarningCollector.NOOP)` in `TestIcebergHiveStatistics.getScanStatsEstimate(...)` will firstly execute the sql, then fetch its corresponding plan. So it would throw an error in java worker when `pushdown_filter_enabled` is true.

Reply:
You were correct about the execution. I think it was an issue with the way I generated the logical plan. The session property was not being propagated properly to the connector (it was always false). I dug up some more examples and found a better way to get the logical plan which doesn't involve executing the query. I also confirmed that the pushdown property is set correctly. Thanks for pointing out the issue